src/profiles/utils.ts

```diff
@@ -0,0 +1,466 @@
+                baseProfile.profile.rejectUnauthorized),
+            basePath: serviceProfile.profile.basePath,
+            hostname: serviceProfile.profile.host ? serviceProfile.profile.host : baseProfile.profile.host,
+            port: serviceProfile.profile.port ? serviceProfile.profile.port : baseProfile.profile.port,
+        };
+
+        const cmdArgs: ICommandArguments = {
+            $0: "zowe",
+            _: [""],
+            tokenType: "apimlAuthenticationToken",
+            tokenValue: baseProfile.profile.tokenValue
+        };
+
+        try {
+            let connectableSessCfg: ISession;
+            if (prompt) {
+                connectableSessCfg = await ConnectionPropsForSessCfg.addPropsOrPrompt<ISession>(sessCfg,
+                    cmdArgs,
+                    {
+                        requestToken: false,
+                        doPrompting: prompt,
+                        getValuesBack: this.collectProfileDetails
+                    });
+            } else {
+                connectableSessCfg = await ConnectionPropsForSessCfg.addPropsOrPrompt<ISession>(sessCfg,
+                    cmdArgs,
+                    { requestToken: false, doPrompting: false });
+            }
+
+            return new Session(connectableSessCfg);
+        } catch (error) {
+            // tslint:disable:no-magic-numbers
+            if (error.mDetails && error.mDetails.errorCode === 401) {
+                if (globals.ISTHEIA) {
+                    vscode.window.showErrorMessage(error.message);
+                    this.getValidSession(serviceProfile, serviceProfile.name, true);
+                } else {
+                    vscode.window.showErrorMessage(error.message, "Check Credentials").then(async (selection) => {
+                        if (selection) {
+                            delete serviceProfile.profile.user;
+                            delete serviceProfile.profile.password;
+                            await this.getValidSession(serviceProfile, serviceProfile.name, true);
+                        }
+                    });
+                }
+            } else { throw error; }
+        }
+    } else {
+        // Neither baseProfile nor serviceProfile exists. It is impossible to login with the currently-provided information.
+        throw new Error(localize("getValidSession.loginImpossible",
+            "Profile {0} is invalid. Please check your login details and try again.", profileName));
+    }
+}
+
+export async function collectProfileDetails(detailsToGet?: string[], oldDetails?: any, schema?: any): Promise<any> {
+    let newUrl: any;
+    let newPort: number;
+    let newUser: string;
+    let newPass: string;
+    let newRU: boolean;
+    const schemaValues: any = {};
+
+    const profileType = "zosmf";
+    if (!profileType) {
+        throw new Error(localize("collectProfileDetails.profileTypeMissing",
+            "No profile type was chosen. Operation Cancelled"));
+    }
```

Comment:
I'm a little confused about what `profileType` is doing here. It's hardcoded with a value of `"zosmf"`, which would normally be a red flag for me, but it seems to only be used for the `if (!profileType)` check right below it, at line 184, which seems like it will always evaluate to `false`. I don't see it being returned or referred to anywhere else in the function. Could we remove it, or is it being used somehow?

Reply:
since it's initialized as "zosmf" in 183 the if statement below with the error will never be used